packages/sdk/src/Actions/EventAction.ts

```diff
@@ -526,21 +531,87 @@ export class EventAction extends DeployableTarget<
    * @public
    * @async
    * @param {ActionStep} actionStep - The action step to validate. Can be a function of event step.
-   * @param {?TxParams & { chainId?: number }} [params] - Additional parameters for validation, including known events, logs, and chain ID.
+   * @param {TxParams & { transactionHash?: Hex, chainId?: number }} params - Additional parameters for validation, including, hash, known events, logs, and chain ID.
    * @returns {Promise<boolean>}
    */
   public async isActionStepValid(
     actionStep: ActionStep,
-    params?: TxParams & { chainId?: number },
+    params: TxParams & { hash?: Hex; chainId?: number },
```

Comment:
i think this `TxParams` type is too much now, since we're not calling `getLogs` anywhere and the `ValidateEventStepParams` type extends `GetLogsParams`.

seems like the type should be
```ts
export type ValidateEventStepParams = {
    knownEvents?: Record<Hex, AbiEvent>;
  } & ({ logs: EventLogs } | { hash: Hex, chainId: number });
```
then we don't realllly need to check if the params has hash and chainId, because it'll be type safe from returning early if params.logs exists.